spec/user_interface_spec.rb

```diff
@@ -22,6 +30,34 @@
     end
   end
 
+  describe 'process_input' do
+    it 'returns the selection if valid' do
+      expect(user_interface.process_input(method(:callback), '2', script.game_modes)).to eql('humanVScomputer')
+    end
+
+    it 'raises an error if selection is not valid' do
+      expect(user_interface.process_input(method(:callback), '4', script.game_modes)).to raise_error('Your selection is not valid.')
+    end
+  end
+
+  describe 'get_input' do
+    it 'puts message to STDOUT' do
+      allow(user_interface).to receive(:gets).and_return('1')
+      expect(STDOUT).to receive(:puts).with('message')
```

Comment:
🙀 

One way to get rid of this mocking is if we removed the hard dependency on `STDOUT` and `puts`. How could we do that?

An approach that I like is to eliminate hard dependencies on anything you don't own. In this case, that would `STDOUT` and `puts`.